src/main/python/image_preprocessing/build_director.py

```diff
@@ -17,45 +17,65 @@ class BuildDirector:
     The BuildDirector constructs the Pipeline using the PipelineBuilder
     """
     @staticmethod
-    def construct(preferences):
+    def construct(preferences, identification_type):
         """
         This function constructs the pipeline
         Author(s):
-            Nicolai van Niekerk
+            Nicolai van Niekerk and Marno Hermann
         Args:
             preferences (dict): User-specified techniques to use in pipeline.
+            identification_type (string): Containts the type of identification, this is used
+                                        to determine which techniques are used.
         Returns:
             :Pipeline (Constructed pipeline)
         """
         builder = PipelineBuilder()
-
         # Use template matching to identify type here
 
         if 'blur_method' in preferences:
             blur_method = preferences['blur_method']
+        elif identification_type['type'] == 'idcard':
```

Comment:
If `identification_type` was a string this would've broken. Why is `identification_type` still a dict?